### ISSUE TITLE
about show_models()

### ISSUE THREAD

**Issue by palapalamao** [automl/auto-sklearn#382]
![image](https://user-images.githubusercontent.com/7733481/32334440-6743c0e8-c025-11e7-9a7c-6fad81691050.png)
I am new to sklearn. which weight is better regressor? and what is the weight mean? how to convert a sklearn code?

**Comment by palapalamao** [comment]
/usr/bin/python3.6 "/home/allan/Desktop/auto ml/tutorials-master/auto-sklearn/example_regression.py"
[WARNING] [2017-11-02 18:41:43,366:AutoMLSMBO(1)::boston] Could not find meta-data directory /usr/local/lib/python3.6/dist-packages/auto_sklearn-0.2.1-py3.6-linux-x86_64.egg/autosklearn/metalearning/files/r2_regression_dense
[WARNING] [2017-11-02 18:42:01,018:smac.intensification.intensification.Intensifier] Challenger was the same as the current incumbent; Skipping challenger
[WARNING] [2017-11-02 18:42:01,018:smac.intensification.intensification.Intensifier] Challenger was the same as the current incumbent; Skipping challenger
[WARNING] [2017-11-02 19:05:16,036:smac.intensification.intensification.Intensifier] Challenger was the same as the current incumbent; Skipping challenger
[WARNING] [2017-11-02 19:05:16,036:smac.intensification.intensification.Intensifier] Challenger was the same as the current incumbent; Skipping challenger
[WARNING] [2017-11-02 19:08:38,848:smac.intensification.intensification.Intensifier] Challenger was the same as the current incumbent; Skipping challenger
[WARNING] [2017-11-02 19:08:38,848:smac.intensification.intensification.Intensifier] Challenger was the same as the current incumbent; Skipping challenger
[WARNING] [2017-11-02 19:22:01,728:smac.intensification.intensification.Intensifier] Challenger was the same as the current incumbent; Skipping challenger
[WARNING] [2017-11-02 19:22:01,728:smac.intensification.intensification.Intensifier] Challenger was the same as the current incumbent; Skipping challenger
[(0.500000, SimpleRegressionPipeline({
'imputation:strategy': 'mean', '
one_hot_encoding:use_minimum_fraction': 'False', 
'preprocessor:__choice__': 'no_preprocessing', 
'regressor:__choice__': 'adaboost', 
'rescaling:__choice__': 'minmax', 
'regressor:adaboost:learning_rate': 0.31003776784191134, 
'regressor:adaboost:loss': 'square', 
'regressor:adaboost:max_depth': 10, 
'regressor:adaboost:n_estimators': 76},
dataset_properties={
  'task': 4,
  'sparse': False,
  'multilabel': False,
  'multiclass': False,
  'target_type': 'regression',
  'signed': False})),
(0.280000, SimpleRegressionPipeline({
'imputation:strategy': 'mean', 
'one_hot_encoding:use_minimum_fraction': 'False', 
'preprocessor:__choice__': 'no_preprocessing', 
'regressor:__choice__': 'adaboost', 
'rescaling:__choice__': 'minmax', 
'regressor:adaboost:learning_rate': 1.6009263315653246, 
'regressor:adaboost:loss': 'exponential', 
'regressor:adaboost:max_depth': 10,
 'regressor:adaboost:n_estimators': 50},
dataset_properties={
  'task': 4,
  'sparse': False,
  'multilabel': False,
  'multiclass': False,
  'target_type': 'regression',
  'signed': False})),
(0.220000, SimpleRegressionPipeline({
'imputation:strategy': 'mean', 
'one_hot_encoding:use_minimum_fraction': 'False', 
'preprocessor:__choice__': 'no_preprocessing', 
'regressor:__choice__': 'adaboost', 
'rescaling:__choice__': 'minmax', 
'regressor:adaboost:learning_rate': 0.8361684970598493, 
'regressor:adaboost:loss': 'exponential', 
'regressor:adaboost:max_depth': 10, 
'regressor:adaboost:n_estimators': 50},
dataset_properties={
  'task': 4,
  'sparse': False,
  'multilabel': False,
  'multiclass': False,
  'target_type': 'regression',
  'signed': False})),
]
R2 score: 0.855964061366

Process finished with exit code 0

**Comment by mfeurer** [comment]
The weight does not necessarily reflect which regressor is the best by itself, but rather which regressor contributes how much to the final ensemble. If you want to find the best regressor, try setting `ensemble_size=1`. In order to understand how the ensemble works, you may want to check the original publication by [Caruana et al.](https://www.cs.cornell.edu/~caruana/ctp/ct.papers/caruana.icml04.icdm06long.pdf). In order for me to understand the question about how to convert to sklearn, you need to explain what you want here. The class you receive is compatible to sklearn in most ways.

**Comment by palapalamao** [comment]
@mfeurer     X, y = sklearn.datasets.load_boston(return_X_y=True)
    feature_types = (['numerical'] * 3) + ['categorical'] + (['numerical'] * 9)
    X_train, X_test, y_train, y_test = \
        sklearn.model_selection.train_test_split(X, y, random_state=1)

    automl = autosklearn.regression.AutoSklearnRegressor(
        time_left_for_this_task=60, per_run_time_limit=30,
        tmp_folder='/tmp/autoslearn_regression_example_tmp',
        output_folder='/tmp/autosklearn_regression_example_out')
    automl.fit(X_train, y_train, dataset_name='boston',
               feat_type=feature_types)
 
    print("automl.show_models()")
    print(automl.show_models())
    predictions = automl.predict(X_test)
    print("R2 score:", sklearn.metrics.r2_score(y_test, predictions))

In the sample, How to find the  most important features???

**Comment by mfeurer** [comment]
I'm not sure how to answer this question. If you want general feature importance, you may want to have a look at models in scikit-learn which can provide you feature importance statistics. Auto-sklearn will do feature selection internally, but there is no easy and straight-forward way to figure out the features it deems important.

**Comment by mfeurer** [comment]
Closing this due to inactivity.
